chainer/utils/sparse.py

```diff
@@ -0,0 +1,73 @@
+import chainer
+from chainer.backends import cuda
+
+
+class SparseCooMatrix(object):
+
+    def __init__(self, data, row, col, shape, use_variable=False):
+        if not (1 <= data.ndim <= 2):
+            raise ValueError('ndim of data must be 1 or 2.')
+        if not (data.ndim == row.ndim == col.ndim):
+            raise ValueError('ndim of data, row and col must be the same.')
+        if len(shape) != 2:
+            raise ValueError('length of shape must be 2.')
+        if not (shape[0] > 0 and shape[1] > 0):
+            raise ValueError('numbers in shape must be greater than 0.')
+        self.data = data
+        if use_variable:
+            self.data = chainer.Variable(self.data)
+        self.row = row
+        self.col = col
+        self.shape = shape  # (row, col)
+
+    def to_dense(self):
+        xp = cuda.get_array_module(self.data)
+        if self.data.ndim == 1:
+            x = xp.zeros(self.shape, dtype=self.data.dtype)
+            x[self.row, self.col] = self.data
+            return x
+        if self.data.ndim == 2:
+            nb = self.data.shape[0]
+            x = xp.zeros((nb, self.shape[0], self.shape[1]),
+                         dtype=self.data.dtype)
+            for i in range(nb):
+                nnz = xp.count_nonzero(self.data[i])
+                x[i, self.row[i, :nnz],
+                  self.col[i, :nnz]] = self.data[i, :nnz]
+            return x
+
+
+def sparse_dense2coo(x, ldnz=None, use_variable=False):
```

Comment:
How about renaming it to `to_coo` to again mimic scipy? Again here too, could we skip `use_variable` ?

Reply:
> How about renaming it to to_coo to again mimic scipy?

I agree, it's more simple.

> Again here too, could we skip use_variable ?

In that case, do we need additional require_grad option?

Reply:
That would even make it more consistent to the Variable. I think it's an good idea.